regparser/tree/xml_parser/reg_text.py

```diff
@@ -73,13 +73,27 @@ def build_tree(reg_xml):
 
     part = reg_xml.xpath('//PART')[0]
 
+    # Build a list of SUBPARTs, then pull SUBJGRPs into that list:
     subpart_xmls = [c for c in part.getchildren() if c.tag == 'SUBPART']
+    subpart_and_subjgrp_xmls = []
     if len(subpart_xmls) > 0:
-        subparts = [build_subpart(reg_part, s) for s in subpart_xmls]
-        tree.children = subparts
+        for subpart in subpart_xmls:
+            subpart_and_subjgrp_xmls.append(subpart)
+            subjgrps = [c for c in subpart.getchildren() if c.tag == 'SUBJGRP']
+            for subjgrp in subjgrps:
+                subpart_and_subjgrp_xmls.append(subjgrp)
```

Comment:
You can get rid of the `if` and outdent this one level (as subpart_xmls will be an empty list, so iterating over it isn't a problem).

Another minor style thing: you can be a bit more concise by either
```python
subpart_and_subjgrp_xmls.extend(c for c in subpart.getchildren() if c.tag == 'SUBJGRP')
```
or keeping `subjgrps`:
```python
subjpart_and_subjgrps.extend(subjgrps)
```

Something else that I'm not sure you want to address here or not: If you look at reg 646, you'll see that `SUBJGRP`s might exist outside of a subpart. May be worth handling now, I'm not sure.